sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -38,15 +58,22 @@ impl Policy for BearerTokenCredentialPolicy {
         request: &mut Request,
         next: &[Arc<dyn Policy>],
     ) -> PolicyResult {
-        let scopes = self
-            .scopes
-            .iter()
-            .map(|s| s.as_str())
-            .collect::<Vec<&str>>();
-        let access_token = self.credential.get_token(&scopes).await?;
-        let token = access_token.token.secret();
+        let access_token = self.access_token.read().await;
+
+        if let Some(token) = &(*access_token) {
+            if token.is_expired(Some(DEFAULT_REFRESH_TIME)) {
+                let mut access_token = self.access_token.write().await;
+                *access_token = Some(self.credential.get_token(&self.scopes()).await?);
+            }
+        } else {
+            let mut access_token = self.access_token.write().await;
+            *access_token = Some(self.credential.get_token(&self.scopes()).await?);
+        }
 
-        request.insert_header(AUTHORIZATION, format!("Bearer {token}"));
+        request.insert_header(
+            AUTHORIZATION,
+            format!("Bearer {}", self.access_token().await.unwrap()),
```

Comment:
@heaths 
After taking your suggestion to refactor `access_token` to now return an `Option<String>`, this is the last "gotcha" that I am stuck on. What is the suitable way to handle the `Option<String>` unwrapping in the `send` method here? I know that it is definitely NOT `unwrap()` 😄
Thanks!

Reply:
See [`Result::ok`](https://doc.rust-lang.org/stable/std/result/enum.Result.html#method.ok).